File: app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.java
package com.wstxda.gsl;

import android.content.Intent;
import android.service.quicksettings.TileService;

public class MusicSearchQuickSettings extends TileService {
    @Override // android.service.quicksettings.TileService
    public void onClick() {
        super.onClick();
        startActivityAndCollapse(new Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH").setFlags(268435456));
    }
}

File: app/src/main/java/com/wstxda/gsl/MusicSearchQuickSettings.kt
package com.wstxda.gsl

import android.annotation.SuppressLint
import android.content.Intent
import android.service.quicksettings.TileService

@Suppress("DEPRECATION")
class MusicSearchQuickSettings : TileService() {
    @SuppressLint("StartActivityAndCollapseDeprecated")
    override fun onClick() {
        super.onClick()
        startActivityAndCollapse(
            Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH").setFlags(
                Intent.FLAG_ACTIVITY_NEW_TASK
            )
        )
    }
}